### PR TITLE
(py)-fenics-dolfinx: version update

### PR DESCRIPTION
**Pull request by garth-wells**
Update to v0.8.0.

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the new version sha256s and checked `pyproject.toml`.

**Comment by tldahlgren** [comment]
Is this PR ready to be merged or should I wait for other maintainer feedback?

**Review by jhale — APPROVED** [review]

**Pull request merged**
